src/scanners.rs

```diff
@@ -604,16 +604,25 @@ pub fn scan_entity(data: &str) -> (usize, Option<CowStr<'static>>) {
     let mut end = 1;
     if scan_ch(&data[end..], b'#') == 1 {
         end += 1;
-        if end < size && (data.as_bytes()[end] == b'x' || data.as_bytes()[end] == b'X') {
+        let next_byte = data.as_bytes()[end];
+        if end < size && (next_byte == b'x' || next_byte == b'X') {
```

Comment:
Woops, this will probably panic now when `end >= size`. I'll revert this change real quick.